go/database/mpt/proof.go

```diff
@@ -115,3 +139,135 @@ func (p WitnessProof) AllStatesZero(root common.Hash, address common.Address, fr
 func (p WitnessProof) AllAddressesEmpty(root common.Hash, from, to common.Address) (tribool.Tribool, error) {
 	panic("not implemented")
 }
+
+// Equals returns true if the two witness proofs are equal.
+func (p WitnessProof) Equals(other WitnessProof) bool {
+	return maps.EqualFunc(p.proofDb, other.proofDb, rlpEncodedNodeEquals)
+}
+
+// MergeProofs merges the input witness proofs into the first witness proof.
```

Comment:
nit: The comment seems to be outdated; the function now supports a list of proofs to be merged, which is nice.

Reply:
fixed